sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClientCallTask.java

```diff
@@ -0,0 +1,337 @@
+                mCallTemplate.onBeforeCalling(url, mHeaders);
+            }
+
+            /* Send payload. */
+            if (binaryPayload != null) {
+
+                /* Log payload. */
+                if (AppCenterLog.getLogLevel() <= Log.VERBOSE) {
+                    if (payload.length() < MAX_PRETTIFY_LOG_LENGTH) {
+                        payload = TOKEN_REGEX_URL_ENCODED.matcher(payload).replaceAll("token=***");
+                        if (CONTENT_TYPE_VALUE.equals(mHeaders.get(CONTENT_TYPE_KEY))) {
+                            payload = new JSONObject(payload).toString(2);
+                        }
+                    }
+                    AppCenterLog.verbose(LOG_TAG, payload);
+                }
+
+                /* Compress payload if large enough to be worth it. */
+                if (shouldCompress) {
+                    ByteArrayOutputStream gzipBuffer = new ByteArrayOutputStream(binaryPayload.length);
+                    GZIPOutputStream gzipStream = new GZIPOutputStream(gzipBuffer);
+                    gzipStream.write(binaryPayload);
+                    gzipStream.close();
+                    binaryPayload = gzipBuffer.toByteArray();
+                }
+
+                /* Send payload on the wire. */
+                urlConnection.setDoOutput(true);
+                urlConnection.setFixedLengthStreamingMode(binaryPayload.length);
+                OutputStream out = urlConnection.getOutputStream();
+
+                //noinspection TryFinallyCanBeTryWithResources
+                try {
+                    writePayload(out, binaryPayload);
+                } finally {
+                    out.close();
+                }
+            }
+            if (isCancelled()) {
+                return null;
+            }
+
+            /* Read response. */
+            int status = urlConnection.getResponseCode();
+            String response = readResponse(urlConnection);
+            if (AppCenterLog.getLogLevel() <= Log.VERBOSE) {
+                String contentType = urlConnection.getHeaderField(CONTENT_TYPE_KEY);
+                String logPayload;
+                if (contentType == null || contentType.startsWith("text/") || contentType.startsWith("application/")) {
+                    logPayload = TOKEN_REGEX_JSON.matcher(response).replaceAll("token\":\"***\"");
+                } else {
+                    logPayload = "<binary>";
+                }
+                AppCenterLog.verbose(LOG_TAG, "HTTP response status=" + status + " payload=" + logPayload);
+            }
+
+            /* Accept all 2xx codes. */
+            if (status >= 200 && status < 300) {
+                return response;
+            }
+
+            /* Generate exception on failure. */
+            throw new HttpException(status, response);
+        } finally {
+
+            /* Release connection. */
+            urlConnection.disconnect();
+        }
+    }
+
+    private boolean isHttpResult(Object result) {
```

Comment:
I would inline this method

Reply:
Ok. Just in some previous state it was used in two places